agrirouter-sdk-dotnet-standard-impl/Service/Messaging/CancellationToken/DefaultCancellationToken.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Threading;
+using Agrirouter.Api.Service.Messaging;
+
+namespace Agrirouter.Impl.Service.Messaging.CancellationToken
+{
+    /// <summary>
+    /// A default implementation, based on tries and waiting time.
+    /// </summary>
+    public class DefaultCancellationToken : ICancellationToken
+    {
+        private readonly int _maxRetries;
+        private readonly int _waitTimeInMilliseconds;
+
+        private int _nrOfRetries = 0;
+
+        public DefaultCancellationToken(int maxRetries, int waitTimeInMilliseconds)
```

Comment:
The parameter sais max**Re**tries, the implementation in FetchMessageService implies that it really is "maxTries". 

If you set maxRetries to 0, it will not be performed at all.